metrics/internal/collectors/storage-cluster.go

```diff
@@ -0,0 +1,88 @@
+package collectors
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	v1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/metrics/internal/options"
+)
+
+type StorageClusterCollector struct {
+	KMSServerConnectionStatus *prometheus.Desc
+	Informer                  cache.SharedIndexInformer
+}
+
+var _ prometheus.Collector = &StorageClusterCollector{}
+
+func NewStorageClusterCollector(opts *options.Options) *StorageClusterCollector {
+	cl, err := GetOcsClient(opts)
+	if err != nil {
+		klog.Errorf("Unable to get client: %v", err)
+		return nil
+	}
+	lw := cache.NewListWatchFromClient(cl, "storageclusters", metav1.NamespaceAll, fields.Everything())
+	sharedIndexInformer := cache.NewSharedIndexInformer(lw, &v1.StorageCluster{}, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	return &StorageClusterCollector{
+		KMSServerConnectionStatus: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "storagecluster", "kms_connection_status"),
+			`KMS Connection Status; 1: Connected, 2: NotConnected, 3: KMS not enabled`,
+			[]string{"name", "namespace"},
+			nil,
+		),
+		Informer: sharedIndexInformer,
+	}
+}
+
+func (c *StorageClusterCollector) Run(stopCh <-chan struct{}) {
+	go c.Informer.Run(stopCh)
+}
+
+func (c *StorageClusterCollector) Describe(ch chan<- *prometheus.Desc) {
+	ds := []*prometheus.Desc{
+		c.KMSServerConnectionStatus,
+	}
+	for _, d := range ds {
+		ch <- d
+	}
+}
+
+func (c *StorageClusterCollector) Collect(ch chan<- prometheus.Metric) {
+	scLister := NewStorageClusterLister(c.Informer.GetIndexer())
+	storageClusters := getAllStorageClusters(scLister)
+	if len(storageClusters) > 0 {
+		c.collectKMSConnectionStatuses(ch, storageClusters)
+	}
+}
+
+func getAllStorageClusters(lister StorageClusterLister) []*v1.StorageCluster {
+	storageClusters, err := lister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("couldn't list StorageCluster: %v", err)
+		return nil
+	}
+	return storageClusters
+}
+
+func (c *StorageClusterCollector) collectKMSConnectionStatuses(ch chan<- prometheus.Metric, storageClusters []*v1.StorageCluster) {
+	for _, storageCluster := range storageClusters {
+		v := 3
+		if storageCluster.Spec.Encryption.Enable || storageCluster.Spec.Encryption.ClusterWide {
+			v = 1
+			if storageCluster.Status.KMSServerConnection.KMSServerConnectionError != "" {
+				v = 2
+			}
+		}
```

Comment:
Changes look good @Nikhil-Ladha 
Just to keep the code consistent with other `collectors` (see file: metrics/internal/collectors/ceph-object-store.go or file: metrics/internal/collectors/rbd-mirror.go, where OK state is ZERO) can we make `Connected` state as ZERO and rest of the states according to your choice (something like 0: connected, 1: not-connected/failure, 2: kms not enabled).

PS: In my previous comment, I was just pointing out three different scenarios, sorry if that confused you... =)

Reply:
Cool, updated the code.